ng-dev/pr/takeover-pr/takeover-pr.ts

```diff
@@ -0,0 +1,72 @@
+import {dirname, join} from 'path';
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client.js';
+import {Log, bold, green} from '../../utils/logging.js';
+import {checkOutPullRequestLocally} from '../common/checkout-pr.js';
+import {fileURLToPath} from 'url';
+
+/** List of accounts that are supported for takeover. */
+const takeoverAccounts = ['angular-robot'];
+
+export async function takeoverPullRequest({
+  pr,
+  branch,
+}: {
+  pr: number;
+  branch?: string;
+}): Promise<void> {
+  /** An authenticated git client. */
+  const git = await AuthenticatedGitClient.get();
+  /** The branch name used for the takeover change. */
+  const branchName = branch ?? `pr-${pr}`;
+
+  // Make sure the local repository is clean.
+  if (git.hasUncommittedChanges()) {
+    Log.error(
+      ` ✘ Local working repository not clean. Please make sure there are no uncommitted changes`,
+    );
+    return;
+  }
+
+  // Verify that the expected branch name is available.
+  if (git.runGraceful(['rev-parse', '-q', '--verify', branchName]).status === 0) {
+    Log.error(` ✘ Expected branch name \`${branchName}\` already in use`);
+    return;
+  }
+
+  /** The pull request information from Github. */
+  const {data: pullRequest} = await git.github.pulls.get({pull_number: pr, ...git.remoteParams});
+
+  // Confirm that the takeover request is being done on a valid pull request.
+  if (!takeoverAccounts.includes(pullRequest.user.login)) {
```

Comment:
Rather than takeover a pr like this, you should just push the changes to the original pull request. This is mostly existing for the case where we don't have access to this (such as all of the github-robot pull requests).

Reply:
Nit: I'd prefer not limiting, but maybe prompting/warning if it's a different account. This seems useful.